packages/block-library/src/query/edit/inspector-controls/index.js

```diff
@@ -247,6 +247,7 @@ export default function QueryInspectorControls( {
 								onDeselect={ () => setQuerySearch( '' ) }
 							>
 								<TextControl
+									__nextHasNoMarginBottom
```

Comment:
To note, there is an additional margin override, but it appears to be coming from `ToolsPanel`: 

https://github.com/WordPress/gutenberg/blob/fae9da1723260bc8e7375918086423734a76f3ec/packages/components/src/tools-panel/styles.ts#L120